docs/architecture/adr-073-libp2p.md

```diff
@@ -0,0 +1,189 @@
+# ADR 073: Adopt LibP2P
+
+## Changelog
+
+- 2021-11-02: Initial Draft (@tychoish)
+
+## Status
+
+Proposed.
+
+## Context
+
+As part of the 0.35 development cycle, the Tendermint team completed
+first phase of the work described in ADRs 61 and 62, which included a
+large scale refactoring of the reactors and the p2p message
+routing. This replaced the switch and many of the other legacy
+components, without breaking protocol or network-level
+interoperability and left the legacy connection/socket handling code.
+
+Following the release, the team has reexamined the state of the code
+and the design, as well as Tendermints requirements for. The notes
+from this process are available in the [P2P Roadmap
```

Comment:
```suggestion
from that process are available in the [P2P Roadmap
```

Minor nit only to distinguish "this" meaning this document from "this" meaning the previous document referenced.